code/code/disc/disc_murder.cc

```diff
@@ -137,34 +137,33 @@ int TBeing::backstabHit(TBeing *victim, TThing *obj)
               FALSE, this, obj, victim, TO_NOTVICT);
         }
 
-        act("Suddenly, $n stabs you in the back!",
-            FALSE, this, obj, victim, TO_VICT);
-
-	// poison
-	TBaseWeapon *tow;
-	if(obj && (tow = dynamic_cast<TBaseWeapon *>(obj)) && 
-	   tow->isPoisoned())
-	  tow->applyPoison(victim);
+        act("Suddenly, $n stabs you in the back!", FALSE, this, obj, victim, TO_VICT);  
+
+        auto weapon = dynamic_cast<TBaseWeapon *>(obj);
+        
+        if (weapon && weapon->checkSpec(victim, CMD_BACKSTAB, "-special-", this) == DELETE_VICT) {
+          delete victim;
+          victim = NULL;
+          return DELETE_VICT;
+        }
 
+        // poison        
+        if (victim && weapon && weapon->isPoisoned())
+          weapon->applyPoison(victim);     
       }
     }
   } else {
-    act("$N quickly avoids your backstab, and you nearly cut your finger.",
-        FALSE, this, obj, victim, TO_CHAR);
-    act("$n tried to backstab you, but you avoid $m.",
-        FALSE, this, obj, victim, TO_VICT);
-    act("$n tried to backstab $N, but nearly cut $s own finger.",
-        FALSE, this, obj, victim, TO_NOTVICT);
-
+    act("$N quickly avoids your backstab, and you nearly cut your finger.", FALSE, this, obj,
+        victim, TO_CHAR);
+    act("$n tried to backstab you, but you avoid $m.", FALSE, this, obj, victim, TO_VICT);
+    act("$n tried to backstab $N, but nearly cut $s own finger.", FALSE, this, obj, victim,
+        TO_NOTVICT);
     d = 0;
   }
 
   if (reconcileDamage(victim, d, SKILL_BACKSTAB) == -1)
     return DELETE_VICT;
 
-  victim->addHated(this);
```

Comment:
Did this get moved somewhere else or did you find in not needed? What's the purpose in removing it?

Reply:
Yeah, it actually gets called almost immediately after this function returns, at line 354, so this call was redundant.